packages/app/src/background/services/validation/index.ts

```diff
@@ -18,3 +20,12 @@ export const validateZkInputs = (payload: Required<ZkInputs>): Required<ZkInputs
 
   return payload;
 };
+
+export const validateVerifiableCredential = async (serializedVerifiableCredential: string): Promise<string> => {
```

Comment:
Agree with duplication. If it's possible to have less code - the better.

Reply:
I've refactored the validation logic as suggested - the `validateVerifiableCredential` function now takes in a string and just does validation. I also deleted the other copy of the function. That being said, I don't think it makes sense to separate validation from `deserializeVerifiableCredential`, as yup is doing both validation and deserialization simultaneously, i.e. transforming dates and such. As such, I've renamed `validateVerifiableCredential` to `validateSerializedVerifiableCredential` since essentially it is just validating strings, which is what is needed for the handler check.